src/main/java/com/ss/mqtt/broker/service/impl/SimpleAuthenticationService.java

```diff
@@ -0,0 +1,41 @@
+package com.ss.mqtt.broker.service.impl;
+
+import com.ss.mqtt.broker.exception.ConnectionRejectException;
+import com.ss.mqtt.broker.model.ConnectAckReasonCode;
+import com.ss.mqtt.broker.network.packet.in.ConnectInPacket;
+import com.ss.mqtt.broker.service.AuthenticationService;
+import com.ss.mqtt.broker.service.ClientIdRegistry;
+import com.ss.mqtt.broker.service.CredentialSource;
+import com.ss.rlib.common.util.StringUtils;
+import lombok.RequiredArgsConstructor;
+import org.jetbrains.annotations.NotNull;
+import reactor.core.publisher.Mono;
+
+@RequiredArgsConstructor
+public class SimpleAuthenticationService implements AuthenticationService {
+
+    private final @NotNull CredentialSource credentialSource;
+    private final @NotNull ClientIdRegistry clientIdRegistry;
+    private final boolean allowAnonymousAuth;
+
+    @Override
+    public @NotNull Mono<String> auth(@NotNull ConnectInPacket packet) {
+        if (allowAnonymousAuth && !packet.isHasUserName()) {
```

Comment:
you can use nullable username and password instead of passing a packet as argument